src/cmd-koji-upload

```diff
@@ -195,20 +196,41 @@ class Build(_Build):
 
         return fname, True
 
-    def get_rpm_list(self, host=None):
+    def get_rpm_list(self, **kwargs):
         """
         Translate commitmeta.json/HOST OS rpms into a json list
         Returns the json rpms list
         """
         components = []
-        if host is None:
+        if "meta" in kwargs:
+            file_path = f"{self.buildroot}/meta.json"
+            with open(file_path) as f:
+                data = json.load(f)
+                rpms = data['rpmdb.pkglist']
+        if "extensions" in kwargs:
+            file_path = f"{self.buildroot}/extensions.json"
+            with open(file_path) as f:
+                 data = json.load(f)
+                 rpms = []
+                 # Extensions in node image are stored different
+                 for name, full_version in data.items():
+                     version_release, arch = full_version.rsplit('.', 1)
+                     version_str, release = version_release.rsplit('-', 1)
+                     if '.' in version_str:
+                         epoch, version = version_str.split('.', 1)
+                     else:
+                         epoch = '0'
+                         version = version_str
+                     rpm_list = [name, epoch, version, release, arch]
+                     rpms.append(rpm_list)
+        if "commitmeta" in kwargs:
             rpms = self.commit["rpmostree.rpmdb.pkglist"]
-        else:
+        if "host" in kwargs:
```

Comment:
All the top-level if-statements other than the first should be `elif` instead, no?